src/utilities/util_verify.c

```diff
@@ -61,22 +61,29 @@ util_verify(WT_SESSION *session, int argc, char *argv[])
     argc -= __wt_optind;
     argv += __wt_optind;
 
-    /* The remaining argument is the table name. */
-    if ((argc != 1 && !hs_verify) || (hs_verify && argc != 0))
-        return (usage());
-    if (!hs_verify && ((uri = util_uri(session, *argv, "table")) == NULL))
-        return (1);
+    /*
+     * The remaining argument is the table name. If we are verifying the history store we do not
+     * accept a URI. Otherwise, we need a URI top operate on.
+     */
+    if (hs_verify && argc != 0)
+        (void)util_err(session, 0, "-a can't be used along with a uri");
```

Comment:
I'm confused why we need `-a`, I'm sure I'm missing something obvious, but couldn't we verify the `file:WiredTigerHS.wt` uri?

Reply:
At the moment verifying ```file:WiredTigerHS.wt``` will attempt to verify the internal contents of the history store file and fail at trying to obtain the lock. There is a ticket open to find a way to get past the exclusive access requirement, WT-5352.

The idea behind having a separate flag/option is that we can add more history-store specific checks behind that flag and not just limit to file verification. Also, it is cleaner to call it through API, as a configuration option, "history-store":true, instead of URI as "file:WiredTigerHS.wt".

Alternatively, we can strcmp the passed URI with "file:WiredTigerHS.wt" and do additional checks if there is a match, but I think passing an API option is cleaner for the end-user.

Reply:
I disagree, but I won't fight to the death on this one. To lay out the case:

* First, this is a debugging option, end-users won't ever call it and it's a rare option for even developers, so I'm pretty comfortable not having a shorthand to get to the functionality. End-users don't actually care if the history file is verifiable, as long as it delivers the correct results for the other files in the system, with luck they won't even be aware of it.

* Second, as you say, we can do whatever we like by looking at the URI, and in fact, that's what we're doing for all URIs, by looking at the history store as well as the passed in URI argument. So, adding an option doesn't add functionality. (We've didn't do anything like this for the metadata file, or the lookaside file, I'm not sure how the use cases change for the durable history file.)

* Finally, there's a limit to the verification we can do without an exclusive lock: we need to think a bit about how this changes the meaning of "verification". If we do go down that route, perhaps we should offer a "verification light" for all verifications, not just the durable history file.

Reply:
"file:WiredTigerHS.wt" is an internal name, my preference is to hide it when specified as part of the API. If you disagree we can continue that discussion in WT-5662.